src/module.c

```diff
@@ -3043,7 +3067,8 @@ int RM_HashSet(RedisModuleKey *key, int flags, ...) {
 
         robj *argv[2] = {field,value};
         hashTypeTryConversion(key->value,argv,0,1);
-        updated += hashTypeSet(key->value, field->ptr, value->ptr, low_flags);
+        int updated = hashTypeSet(key->value, field->ptr, value->ptr, low_flags);
+        count += (flags & REDISMODULE_HASH_COUNT_INSERTS) ? 1 : updated;
```

Comment:
don't we need to update errno here too? in case `updated` is 0

Reply:
updated == 0 means it was inserted. That's not an error.

The comment above hashTypeSet() says:

    /* Add a new field, overwrite the old with the new value if it already exists.
     * Return 0 on insert and 1 on update.

Reply:
i meant that since the API doc says that errno will be set if the return value is less than expected.
so when we don't increment `count` here, the user may check `errno` and it might be a leftover from another function.

Reply:
Yes, currently the only way to know is to use the COUNT_INSERTS flag.

So you're saying we should set it to ENOENT if all fields were inserted and COUNT_INSERTS was not set?

Reply:
i'm saying that if we state that errno is set when the return value (count) is less than the input count, then we must set it in any case were we do `continue` or avoid incrementing the return count.

if we state that errno is only set when the return value is 0, then i think that's easier, and IIRC that was the main problem (distinguish between errors and cases were the count is 0 without an error)
in that case we can just set it to some value indicating success.
like: if you got 0 as return and errno is ENOENT, then that's not an error.. other errno values with 0 (e.g. EINVAL) indicate an error.

Reply:
Yes, I get it and I've updated the PR as you suggested: Only setting errno when returning 0.

> if we state that errno is set when the return value (count) is less than the input count

Well, that's why I wrote "less than expected" (implying that it could be less than the input when inserts are not counted) but it's way too confusing.

See the added fixup commit, very small.